cmake/thirdparty/SetupMPI.cmake

```diff
@@ -183,10 +183,13 @@ if (ENABLE_FORTRAN)
     endif()
 endif()
 
-blt_import_library(NAME          mpi
-                   INCLUDES      ${_mpi_includes}
-                   TREAT_INCLUDES_AS_SYSTEM ON
-                   LIBRARIES     ${_mpi_libraries}
-                   COMPILE_FLAGS ${_mpi_compile_flags}
-                   LINK_FLAGS    ${_mpi_link_flags} 
-                   EXPORTABLE    ${BLT_EXPORT_THIRDPARTY})
+# Don't create the mpi target if it already exists.
+if (NOT TARGET blt::mpi)
+    blt_import_library(NAME          mpi
```

Comment:
@white238 @cyrush Can either of you think of a case where we might need to patch this target if it already exists?

Reply:
not 100% sure, I think there is an overall order dependence with the new setup that I don't grasp enough to know how that impacts this. 
 
Here we create `mpi` if `blt::mpi` inst defined, but would expect we also need to define the `blt::mpi` target here if not defined to keep the logic straight?